model/src/main/java/org/projectnessie/model/Operation.java

```diff
@@ -56,9 +65,20 @@
     @NotNull
     Contents getContents();
 
+    @Nullable
+    Contents getExpectedContents();
```

Comment:
what does this do?

Reply:
That's the `if` for the global-state-CAS-operation (think: "UPDATE global_state SET value = contents.table_metadata IF value = expected_contents.table_metadata")

Reply:
so if null it doesn't check? or if null and set it will fail?

Mildly annoying that we have to send this whole object back but prob no easy way around it. Thoughts?

Reply:
Yea, it's the easiest way for tools: just send back what you got. Otherwise we'd have to introduce a new type just for the global-state + conversion/comparison functionality.

If it's null, it doesn't check, which can be used like a "git push --force". It must be null for non-global-state-content-types (Deltalake) and also when you've got no previous global-state (think: a new table). Sounds more complicated than it actually is (it's really just passing the existing field `NessieTableOperations.table` as the expectedValue).

Reply:
Can we add some javadoc somewhere to that effect? I think it would be weird to reject a DeltaLake op if its non-null and have no clear reason why.

Reply:
I've added a TODO to re-think that behavior in the `PersistVersionStore` (no PR yet)